dequeue-question/index.js

```diff
@@ -0,0 +1,52 @@
+const { QueueServiceClient, StorageSharedKeyCredential } = require("@azure/storage-queue");
+require("dotenv").config();
+module.exports = async function (context, req) {
+  context.log('JavaScript HTTP trigger function processed a request.');
+
+  if (req.query.name || (req.body && req.body.name)) {
+    context.res = {
+      status: 400,
+      body: "no request parameters expected"
+    };
+
+
+  }
+  else {
+    context.res = {
+      body: await dequeue(context)
+    };
+  }
+};
+
+
+
+async function dequeue(context) {
```

Comment:
This is our most complex function so it's probably worth having unit tests for this?